packages/shared/src/utils/browser/browser_detect.ts

```diff
@@ -13,3 +13,27 @@ export const isSafari = () => {
         })(!window.safari || (typeof window.safari !== 'undefined' && window.safari.pushNotification))
     );
 };
+
+const getUserBrowser = () => {
+    // We can't rely only on navigator.userAgent.index, the verification order is also important
+    if ((navigator.userAgent.indexOf('Opera') || navigator.userAgent.indexOf('OPR')) !== -1) {
+        return 'Opera';
+    } else if (navigator.userAgent.indexOf('Edg') !== -1) {
+        return 'Edge';
+    } else if (navigator.userAgent.indexOf('Chrome') !== -1) {
+        return 'Chrome';
+    } else if (navigator.userAgent.indexOf('Safari') !== -1) {
+        return 'Safari';
+    } else if (navigator.userAgent.indexOf('Firefox') !== -1) {
+        return 'Firefox';
+    }
+    return 'unknown';
+};
+
+export const user_browser = {
```

Comment:
Do we need this whole logic here if we just use it to check if it's safari or not? I think Opera, Edge and Chrome should never have any specific logic since they are all based on the same engine.